### PR TITLE
Bug 1795707: Conditionally render DrawerPanelContent when drawer is closed

### PR DESCRIPTION
**Pull request by jcaianirh**
Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=1795707

### CODE REVIEW

File: frontend/packages/patternfly/src/components/notification-drawer/notification-drawer.tsx [hunk 1]
```diff
@@ -17,8 +17,9 @@ const NotificationDrawer: React.FC<NotificationDrawerProps> = ({
     <Drawer isExpanded={isExpanded} isInline={isInline}>
       <DrawerContent>{children}</DrawerContent>
       <DrawerPanelContent noPadding>
-        <NotificationDrawerHeading>{notificationEntries}</NotificationDrawerHeading>
+        {isExpanded && <NotificationDrawerHeading>{notificationEntries}</NotificationDrawerHeading>}
```
Comment (spadgett) [inline frontend/packages/patternfly/src/components/notification-drawer/notification-drawer.tsx]: Don't we want to skip rendering the entire drawer if it's not open? Not just the sections?

**Comment by spadgett** [comment]
I think we should address the way this blocks loading the entire app until ClusterVersion loads as well. We shouldn't block. Can I ask why the drawer wraps the entire app contents? The problem is that any error in the drawer will prevent the entire app from loading as well as having performance implications.

**Comment by spadgett** [comment]
We shouldn't make `NotificationDrawer` an `AsyncComponent` because it will always load.

**Comment by jcaianirh** [comment]
@spadgett the drawer component won't pass the analyze test unless it is loaded asynchronously.

**Comment by jcaianirh** [comment]
@spadgett the drawer wraps the app contents because the drawer uses it as the container which the drawer panel will slide into or over. See: https://www.patternfly.org/v4/documentation/react/experimental/drawer

**Comment by spadgett** [comment]
> @spadgett the drawer component won't pass the analyze test unless it is loaded asynchronously.

This is cheating :) We're basically tricking analyze since it will always load. We should bump the limit if we need to. How much does this add to our vendor bundle?

File: frontend/packages/patternfly/src/components/notification-drawer/notification-drawer.tsx (hunk 1 above)
Comment (jcaianirh) [reply frontend/packages/patternfly/src/components/notification-drawer/notification-drawer.tsx]: @spadgett we want the notification drawer to be active to allow for collecting notifications for the badge icon for both alerts and cluster updates. So we need the cluster version to load for that reason. However, I don't want it to block the app from loading...do you have a suggestion for a workaround for that situation?

**Comment by spadgett** [comment]
> @spadgett the drawer wraps the app contents because the drawer uses it as the container which the drawer panel will slide into or over. See: https://www.patternfly.org/v4/documentation/react/experimental/drawer

That's fair, but can't block loading waiting on Firehose. We need to load the app content immediately. Is it possible to wrap one of the child drawer components using Firehose instead of putting it around the entire drawer?

**Comment by jcaianirh** [comment]
@spadgett I don't think we were far of re the size....I believe it was 3.1 vs the limit which is 3.0. As an aside, I'm looking at maybe 5 components left from pf3...if we can remove them, couldn't we remove the dependency on pf3 and provide a savings in the size and pass the analyze test?

**Comment by jcaianirh** [comment]
@spadgett i will investigate moving the firehose to another level deeper in tree that doesn't block the app. thanks for bringing this to my attention.

**Comment by spadgett** [comment]
@jcaianirh kubevirt-web-ui-components depends on PF3, so we can't remove it yet. we're trying... :)

The limit is 2MB I believe.

**Comment by benjaminapetersen** [comment]
I'm looking at maybe 5 components left from pf3...if we can remove them...

Depends on the size of work and if we can call that a bug.

File: frontend/packages/patternfly/src/components/notification-drawer/notification-drawer.tsx (hunk 1 above)
Comment (spadgett) [reply frontend/packages/patternfly/src/components/notification-drawer/notification-drawer.tsx]: That's fine, but it looks like you're rendering everything in the drawer except the heading here.

Comment (spadgett) [reply frontend/packages/patternfly/src/components/notification-drawer/notification-drawer.tsx]: @jcaianirh Sure, but don't you want to hide the entire `DrawerPanelContent` not just the heading?

I think removing the `AsyncComponent` should make performance a lot better.

Comment (spadgett) [reply frontend/packages/patternfly/src/components/notification-drawer/notification-drawer.tsx]: I guess hiding `DrawerPanelContent` breaks the layout :(

I think the real issue is the `AsyncComponent`, which slows down page load quite a bit

**Comment by spadgett** [comment]
Closing for now in favor of #4192. We can revisit this later, but the general performance problem has been fixed.

/close

**Pull request closed without merging**